code/physioconv/test.py

```diff
@@ -0,0 +1,170 @@
+import re
+import pandas as pd
+
+TRIAL_TYPE = {
+    "bh_body_2": "hold",
+    "bh_body": "hold-test",
+    "bh_end": "hold-end-test",
+    "bh_end_2": "hold-end",
+    "bh_end_3": "breath-freely",
+    "eye_movement_fixation": "cog",
+    "ft_hand": "mot",
+    "fixation": "blank",
+    "grating": "vis",
+    "movie": "movie",
+    "polygon_4": "breath-in",
+    "polygon_5": "breath-out",
+    "polygon_6": "breath-in-last",
+    "polygon_8": "breath-out-last",
+    "polygon_7": "breath-out-last",
+}
+
+file = """
+62.3562 	EXP 	text_2: autoDraw = True
+100.7936 	DATA 	Keypress: s
+100.8108 	EXP 	text_2: autoDraw = False
+100.8108 	EXP 	directions: autoDraw = True
+102.3922 	DATA 	Keypress: s
+281.6250 	EXP 	Imported xpos_ypos.csv as conditions, 6 conditions, 2 params
+281.6253 	EXP 	Created sequence: random, trialTypes=6, nReps=1, seed=None
+281.6255 	EXP 	New trial (rep=0, index=0): OrderedDict([('xpos', -0.7), ('ypos', 0.0)])
+281.6300 	EXP 	grating: autoDraw = False
+281.6300 	EXP 	polygon: autoDraw = False
+281.6300 	EXP 	grating: autoDraw = False
+281.6300 	EXP 	polygon: autoDraw = False
+281.6300 	EXP 	eye_movement_fixation: autoDraw = True
+281.6300 	EXP 	eye_movement_fixation_inner: autoDraw = True
+"""
+
+trigger_pattern = r"([\d.]+)\s+DATA\s+Keypress:\s+s"
+trigger_match = float(re.findall(trigger_pattern, file)[0])
+
+print(trigger_match)
+
+log = "/home/cprovins/data/hcph-pilot/Psychopy_organized/session-2023-07-14/control_task_1_FINAL_2023-07-14_18h27.05.674.log"
+
+with open(log, "r") as f:
+    file = f.read()
```

Comment:
Feels cleaner with pathlib. Also, having all that code inside the context seems unjustified.
```suggestion
log_text = Path(log).read_text()
```